OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -143,24 +184,37 @@ public void Repulse()
 			if (repulsionFacing == -1)
 				return;
 
-			var speed = info.RepulsionSpeed != -1 ? info.RepulsionSpeed : MovementSpeed;
+			var speed = Info.RepulsionSpeed != -1 ? Info.RepulsionSpeed : MovementSpeed;
 			SetPosition(self, CenterPosition + FlyStep(speed, repulsionFacing));
 		}
 
 		public virtual WVec GetRepulsionForce()
 		{
-			if (!info.Repulsable)
+			if (!Info.Repulsable)
 				return WVec.Zero;
 
 			// Repulsion only applies when we're flying!
 			var altitude = CenterPosition.Z;
-			if (altitude != info.CruiseAltitude.Length)
+			if (altitude != Info.CruiseAltitude.Length)
 				return WVec.Zero;
 
-			return self.World.FindActorsInCircle(self.CenterPosition, info.IdealSeparation)
-				.Where(a => !a.IsDead && a.HasTrait<Aircraft>() && a.Info.Traits.Get<AircraftInfo>().CruiseAltitude == info.CruiseAltitude)
+			var repulsionForce = self.World.FindActorsInCircle(self.CenterPosition, Info.IdealSeparation)
+				.Where(a => !a.IsDead && a.HasTrait<Aircraft>() && a.Info.Traits.Get<AircraftInfo>().CruiseAltitude == Info.CruiseAltitude)
 				.Select(GetRepulsionForce)
 				.Aggregate(WVec.Zero, (a, b) => a + b);
+
+			if (Info.CanHover)
+				return repulsionForce;
+
+			var currentDir = FlyStep(Facing);
```

Comment:
Can we get a comment here explaining that non-hovering aircraft must always move forward.